### PR TITLE
Routes: define interface for generic params

### PR DESCRIPTION
**Pull request by barnjamin**
Improve typing on generic route params

### CODE REVIEW

File: connect/src/routes/tokenBridge/automatic.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ type R = TransferReceipt<"AutomaticTokenBridge">;
 export class AutomaticTokenBridgeRoute<N extends Network> extends AutomaticRoute<N, Op, R, Q> {
   NATIVE_GAS_DROPOFF_SUPPORTED = true;
 
-  static getDefaultOptions(): Op {
+  getDefaultOptions(): Op {
```
Comment (barnjamin) [inline connect/src/routes/tokenBridge/automatic.ts:46]: removed static so the same Op param we pass in generics is used

File: connect/src/routes/route.ts [hunk 2]
```diff
@@ -14,19 +16,54 @@ export type ValidationResult<OP> =
...
+    token: TokenId;
+    amount: bigint;
+  };
+  relayerFee?: {
+    token: TokenId;
+    amount: bigint;
+  };
+  // swap slippage? what else?
+  // native gas dropoff?
+  // estimate of _time_?
```
Comment (barnjamin) [inline connect/src/routes/route.ts:46]: 👀

File: connect/src/routes/route.ts [hunk 3]
```diff
@@ -14,19 +16,54 @@ export type ValidationResult<OP> =
...
+}
+
+export interface Quote {
+  sourceToken: {
+    token: TokenId;
+    amount: bigint;
+  };
+  destinationToken: {
+    token: TokenId;
+    amount: bigint;
```
Comment (artursapek) [inline connect/src/routes/route.ts:38]: do we need to complicate this any further for swaps? swaps usually will give a minimum amount (worst case), and an estimated amount. I suppose they could just add a `minimumAmount` field to their quote, right? should we add a `minimumAmount?: bigint` here? wdyt

File: connect/src/routes/route.ts [hunk 4]
```diff
@@ -14,19 +16,54 @@ export type ValidationResult<OP> =
...
-// OP - Options passed to the route
-// R - Reciept of the transfer
-// Q - Quote for the transfer given the options
-export abstract class Route<N extends Network, OP, R, Q> {
+export type UnknownRoute<
+  N extends Network,
+  OP extends Options = Options,
+  R extends Receipt = Receipt,
+  Q extends Quote = Quote,
+> = Route<N, OP, R, Q>;
```
[the remark this answers is not in the document]
Comment (artursapek) [reply connect/src/routes/route.ts:59]: or `<WO, RM, HO, LE>` lol

**Pull request merged**